src/coreclr/vm/finalizerthread.cpp

```diff
@@ -259,6 +259,23 @@ void FinalizerThread::FinalizeAllObjects()
     FireEtwGCFinalizersEnd_V1(count, GetClrInstanceId());
 }
 
+void FinalizerThread::RaiseShutdownEvents()
+{
+    WRAPPER_NO_CONTRACT;
+    fQuitFinalizer = TRUE;
+#ifndef TARGET_WASM
```

Comment:
```suggestion
#ifdef TARGET_WASM
    AppDomain::RaiseExitProcessEvent();
#else    
```
Should this raise shutdown events synchronously? Are shutdown events raised by Mono Wasm?

Reply:
>  Are shutdown events raised by Mono Wasm?

@radekdoulik or @pavelsavara Do you know the answer to this?

Reply:
So `hEventFinalizer` is semaphore which is blocking the finalizer thread when there is no work. This would not be possible on wasm. Instead of `hEventFinalizer->Set();` we will schedule JS timer and instead of `RaiseShutdownEvents` we will [clearTimeout](https://developer.mozilla.org/en-US/docs/Web/API/Window/clearTimeout)